### PR TITLE
Clone repository as provided user/group.

### PR DESCRIPTION
**Pull request by MorganEPatch**
Related to https://github.com/osuosl-cookbooks/timesync/pull/10, the git resource currently clones the repository as the user which calls chef-client (i.e. usually root). This change causes it to check it out as the provided user and group, ensuring proper permissions regardless of caller.

@osuosl-cookbooks/staff

### CODE REVIEW

File: spec/unit/default_spec.rb [hunk 1]
```diff
@@ -30,30 +31,36 @@
   it 'checks out test_a and test_b' do
     expect(chef_run).to sync_git('/opt/custom').with(
       repository: 'https://github.com/osuosl/nodejs-test-apps.git',
-      branch: 'custom')
+      branch: 'custom'
+    )
```
Comment (armiller) [inline spec/unit/default_spec.rb:37]: You need to check if its going to use the correct user and group in the test here.

**Comment by ramereth** [comment]
I'm getting the following error when I run TK:

```
================================================================================
Error executing action `run` on resource 'execute[pm2 startOrReload /etc/pm2/conf.d/test_b.json]'
================================================================================

Mixlib::ShellOut::ShellCommandFailed
------------------------------------
Expected process to exit with [0], but received '8'
---- Begin output of pm2 startOrReload /etc/pm2/conf.d/test_b.json ----
STDOUT: 
STDERR: fs.js:654
  return binding.mkdir(pathModule._makeLong(path),
                 ^
Error: ENOENT, no such file or directory '/home/root/.pm2'
    at Object.fs.mkdirSync (fs.js:654:18)
    at Object.CLI.pm2Init (/usr/lib/node_modules/pm2/lib/CLI.js:40:8)
    at Object.<anonymous> (/usr/lib/node_modules/pm2/bin/pm2:18:5)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:935:3
---- End output of pm2 startOrReload /etc/pm2/conf.d/test_b.json ----
Ran pm2 startOrReload /etc/pm2/conf.d/test_b.json returned 8

Cookbook Trace:
---------------
/tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:41:in `run_action'
/tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:41:in `run_action'
/tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:41:in `run_action'

Resource Declaration:
---------------------
# In /tmp/kitchen/cookbooks/pm2/providers/application.rb

125:   execute cmd do
126:     command cmd
127:     user new_resource.user
128:     environment pm2_environment
129:   end
130: end

Compiled Resource:
------------------
# Declared in /tmp/kitchen/cookbooks/pm2/providers/application.rb:125:in `pm2_command'

execute("pm2 startOrReload /etc/pm2/conf.d/test_b.json") do
  action [:run]
  retries 0
  retry_delay 2
  default_guard_interpreter :execute
  command "pm2 startOrReload /etc/pm2/conf.d/test_b.json"
  backup 5
  environment {"PM2_HOME"=>"/home/root/.pm2"}
  returns 0
  user "root"
  declared_type :execute
  cookbook_name :"nodejs-webapp-test"
end

Platform:
---------
x86_64-linux
```

**Comment by ramereth** [comment]
@LyonesGamer everything looks good here, but would you mind adding the following to the bottom of the metadata.rb? That way we only need to run tests on the platforms we support.

``` ruby
supports         'centos', '~> 6.0'
supports         'centos', '~> 7.0'
```

**Comment by ramereth** [comment]
:+1:

**Comment by MorganEPatch** [comment]
!bump patch ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v1.0.1.  Have a nice day!

**Pull request merged**